full-service/src/validator_ledger_sync.rs

```diff
@@ -145,33 +143,33 @@ impl ValidatorLedgerSyncThread {
                 }
             };
 
-        let blocks_and_contents: Vec<(Block, BlockContents)> = blocks_data
-            .into_iter()
-            .map(|block_data| (block_data.block().clone(), block_data.contents().clone()))
-            .collect();
-
-        mc_ledger_sync::identify_safe_blocks(ledger_db, &blocks_and_contents, logger)
+        mc_ledger_sync::identify_safe_blocks(ledger_db, &blocks_data, logger)
     }
 
-    fn append_safe_blocks(
-        ledger_db: &mut LedgerDB,
-        blocks_and_contents: &[(Block, BlockContents)],
-        logger: &Logger,
-    ) {
+    fn append_safe_blocks(ledger_db: &mut LedgerDB, block_data: &[BlockData], logger: &Logger) {
```

Comment:
Confirming I understand what's going on in this file - this is a refactor due to upstream refactor around using BlockData directly?

Reply:
correct